lib/compfile.stk

```diff
@@ -98,59 +108,72 @@ doc>
         (tm       (clock))
         (copts    (compiler:warn-use-undefined-postpone)))
 
-    ;; Clear the list containing  macros sources
-    (file-module-list-reset!)
-
-    ;; Defer warning til the end of the compilation of file
-    ;(compiler:warn-use-undefined-postpone #t)
-
-    (fluid-let ((*code-instr* '())
-                (*code-constants* '())
-                (*compiling-file* #t))
-
-      ;; SRFI 138, -D option:
-      (when has-prepend?
-        (unless (list? prepend)
-          (error 'compile-file "prepend should be a list: ~S" prepend))
-        ;; whatever the list is, if we wrap it in a BEGIN form,
-        ;; it will be compiled sequentially:
-        (let ((exprs (cons 'begin prepend)))
-          (compile exprs #f exprs #f)))
-
-      ;; Read the source file
-      (%include-file input)
-      (emit 'END-OF-CODE)
-
-      ;; Write the header
-      (format out "#!/usr/bin/env stklos\n")
-      (format out "; A -*- Scheme -*- generated file *DO NOT EDIT**\n")
-      (format out "STklos ~S\n" (compute-file-informations globs))
-
-      (let ((code (assemble (reverse! *code-instr*))))
-        (when (compiler:show-assembly-code)
-          ;;Write the assembly code and code vector
-          (format out "\n#|\n")
-          (disassemble-code code out)
-          (format out "\n~S\n|\#\n" code))
-
-        ;; Write new constants as a vector
-        (format out "#~S\n" *code-constants*)
-
-        ;; Write byte-code
-        (%dump-code out code))
-
-      ;; Close file
-      (close-output-port out))
-
-    ;; Show undefined symbols
-    (compiler-show-undefined-symbols)
-
-    ;; Display time used to compile file
-    (when (and (interactive-port? (current-input-port))
-               (compiler:time-display))
-      (format #t "Compilation time ~S ms\n" (exact (round (- (clock) tm)))))
-
-    ;; Restore Compiler flags
-    (compiler:warn-use-undefined-postpone copts)))
-
-(export compile-file)
\ No newline at end of file
+    ;; If anything goes wrong during compilation, print the error or condition
+    ;; message, remove the output file and exit with a -1 status.
+    (guard (cnd (else (begin
+                        (eprintf "*** Error while compiling: ~a~%" (condition-message cnd))
+                        (eprintf "Backtrace:~%")
+                        (%display-backtrace (condition-ref cnd 'backtrace) 10)
+                        (when (delete-and-exit-on-failure)
```

Comment:
I get an error:
```
**** Error while executing command ("./utils/stklos-compile.stk" "--no-time" "--output=wrong" "wrong.stk")
	 Where: in compile-file
	Reason: bad function `#t'. Cannot be applied
```
I don't know scheme but based on [this post](https://stackoverflow.com/questions/61860245/the-object-f-or-t-is-not-applicable) saying this error can be caused by too many brackets, I tried:
```suggestion
                        (when delete-and-exit-on-failure
```
which seems to help.